test/helper.rb

```diff
@@ -143,3 +143,16 @@ def full_name
     end
   end
 end
+
+module StubConst
+  # Stub constant for the duration of a block
+  def stub_const(name, val, &block)
```

Comment:
Just use `Minitest::Mock`

Reply:
https://github.com/seattlerb/minitest/blob/master/lib/minitest/mock.rb#L212

Reply:
`#stub` doesn't seem to work for constants, only methods.
There is a [`minitest-stub-const`](https://github.com/adammck/minitest-stub-const) extension ([not in core](https://github.com/seattlerb/minitest/pull/230#issuecomment-12869299)) that provides `stub_const`, so I've updated the PR to add that gem-plugin as a dependency if that's preferred to the in-line code.

The alternative would be to replace the `SHUTDOWN_GRACE_TIME` const with a getter-method that can be stubbed with minitest as-is, but I think stubbing the const directly is cleaner. (It  might also be helpful for stubbing other timing-related constants like `WORKER_CHECK_INTERVAL` as well to speed up other tests..)